pkcs11/src/new/types/function.rs

```diff
@@ -217,6 +218,107 @@ pub enum RvError {
+            RvError::RandomSeedNotSupported => write!(f, "This value can only be returned by C_SeedRandom.  It indicates that the token’s random number generator does not accept seeding from an application.  This return value has lower priority than CKR_RANDOM_NO_RNG."),
+            RvError::RandomNoRng => write!(f, "This value can be returned by C_SeedRandom and C_GenerateRandom.  It indicates that the specified token doesn’t have a random number generator.  This return value has higher priority than CKR_RANDOM_SEED_NOT_SUPPORTED."),
+            RvError::DomainParamsInvalid => write!(f, "Invalid or unsupported domain parameters were supplied to the function.  Which representation methods of domain parameters are supported by a given mechanism can vary from token to token."),
+            RvError::BufferTooSmall => write!(f, "The output of the function is too large to fit in the supplied buffer."),
+            RvError::SavedStateInvalid => write!(f, "This value can only be returned by C_SetOperationState.  It indicates that the supplied saved cryptographic operations state is invalid, and so it cannot be restored to the specified session."),
+            RvError::InformationSensitive => write!(f, "The information requested could not be obtained because the token considers it sensitive, and is not able or willing to reveal it."),
+            RvError::StateUnsaveable => write!(f, "The cryptographic operations state of the specified session cannot be saved for some reason (possibly the token is simply unable to save the current state).  This return value has lower priority than CKR_OPERATION_NOT_INITIALIZED."),
+            RvError::CryptokiNotInitialized => write!(f, "This value can be returned by any function other than C_Initialize and C_GetFunctionList.  It indicates that the function cannot be executed because the Cryptoki library has not yet been initialized by a call to C_Initialize."),
+            RvError::CryptokiAlreadyInitialized => write!(f, "This value can only be returned by C_Initialize.  It means that the Cryptoki library has already been initialized (by a previous call to C_Initialize which did not have a matching C_Finalize call)."),
+            RvError::MutexBad => write!(f, "This error code can be returned by mutex-handling functions that are passed a bad mutex object as an argument.  Unfortunately, it is possible for such a function not to recognize a bad mutex object.  There is therefore no guarantee that such a function will successfully detect bad mutex objects and return this value."),
+            RvError::MutexNotLocked => write!(f, "This error code can be returned by mutex-unlocking functions.  It indicates that the mutex supplied to the mutex-unlocking function was not locked."),
+            RvError::NewPinMode => write!(f, "CKR_NEW_PIN_MODE"),
+            RvError::NextOtp => write!(f, "CKR_NEW_PIN_MODE"),
```

Comment:
```suggestion
            RvError::NextOtp => write!(f, "CKR_NEXT_OTP"),
```

Reply:
I'm guessing these two don't have any description in the spec?

Reply:
Arg good catch, will add in the next PR. No they are not even defined... I am not sure where they come from.